src/GameLoveLetter/Cards/BaronCard.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace GameLoveLetter.Cards
+{
+	public class BaronCard : ICard<BaronCardData>
+	{
+		public int Strength { get; set; }
+		public BaronCardData Data { get; set; }
```

Comment:
Why use a nested object ? you can do Effect directly in the card without add complexity

Reply:
In fact, I didn't use a nested object for the `Effect `method, but only to carry information about `CardData` and force child to create their one data. 
See my interface `ICard `and `ICard<CardData> : ICard` for more details. (But I'm not sure my implementation was the best way.)